hanami-controller.gemspec

```diff
@@ -20,7 +20,7 @@ Gem::Specification.new do |spec|
   spec.metadata["rubygems_mfa_required"] = "true"
   spec.required_ruby_version = ">= 3.1"
 
-  spec.add_dependency "rack", "~> 2.0"
+  spec.add_dependency "rack", "~> 3.1"
```

Comment:
I think we'll want to make this `">= 2.0"` like https://github.com/hanami/router/pull/277?

We should add a matrix to CI here to test on both Rack 2 & 3.